storage/storage.c

```diff
@@ -1659,8 +1360,42 @@ static secbool storage_upgrade(void) {
     // Copy all entries.
     uint32_t offset = 0;
     while (sectrue == norcow_get_next(&offset, &key, &val, &len)) {
-      if (sectrue != norcow_set(key, val, len)) {
-        return secfalse;
+      if (NORCOW_VERSION >= 4 && norcow_active_version < 4) {
```

Comment:
The condition `NORCOW_VERSION >= 4` is neither needed now, nor in the future. Is that right?

Reply:
right, should be always true from now on: https://github.com/trezor/trezor-firmware/pull/3228/commits/1cc571c546d6208e76f6fff8e0e168352dabf177